### ISSUE TITLE
Enable networking without the use of proxies (or explain how to set up a NAT proxy to route PalmOS traffic to the host and therefore onto the modern Web)

### ISSUE THREAD

**Issue by TheSystemGuy1337** [cloudpilot-emu/cloudpilot-emu#74]
Just discovered this recently and found out that later Palm devices have a Internet browser, however it requires a proxy. It would be ideal if network support didn't require a proxy or if you flipped the switch, Cloudpilot explains how to set one up or fetches you a default proxy.

**Comment by DirtyHairy** [comment]
The network proxy is required because a web browser cannot open arbitrary sockets, just HTTP (and websocket) requests. The proxy communicates with the emulator via websocket and handles the actual connections. There is no way around this limitation. It is not a standard SOCKS or CONNECT proxy, but a custom protocol that uses protobuf for RPC.

There is a native networking implementation that requires additional support from the runtime environment which is available in the iOS app (which uses a customised webview) and in the native build. With a bit of work, similar shim apps could be developed for Android and the desktop (using Electron or Tauri), but this is currently definitely on the backburner (or a nice project for potential contributors 😛 ).

I have also considered a kind of wizard in the app that takes you through setting up the proxy, but this is a lot of work to do properly (as there are many different configurations and potential issues), so I went for a (hopefully) exhaustive [document](https://github.com/cloudpilot-emu/cloudpilot-emu/blob/master/doc/networking.md). Again, I would be more than happy to accept contributions here.

**Comment by TheSystemGuy1337** [comment]
> The network proxy is required because a web browser cannot open arbitrary sockets, just HTTP (and websocket) requests. The proxy communicates with the emulator via websocket and handles the actual connections. There is no way around this limitation. It is not a standard SOCKS or CONNECT proxy, but a custom protocol that uses protobuf for RPC.
> 
> There is a native networking implementation that requires additional support from the runtime environment which is available in the iOS app (which uses a customised webview) and in the native build. With a bit of work, similar shim apps could be developed for Android and the desktop (using Electron or Tauri), but this is currently definitely on the backburner (or a nice project for potential contributors 😛 ).
> 
> I have also considered a kind of wizard in the app that takes you through setting up the proxy, but this is a lot of work to do properly (as there are many different configurations and potential issues), so I went for a (hopefully) exhaustive [document](https://github.com/cloudpilot-emu/cloudpilot-emu/blob/master/doc/networking.md). Again, I would be more than happy to accept contributions here.

I'm on PC. It would be nice if it did walk you through setting up a proxy when you first flip the switch.

**Comment by TheSystemGuy1337** [comment]
I tried setting up the proxy but can't seem to connect to anything. Not even WAP websites.

**Comment by DirtyHairy** [comment]
No, you're not an idiot 😏 The proxy enables you to connect to the internet with PalmOS, and you can use this to do network hotsync, FTP, VNC and anything else you'd like to do, including surfing the web. However, web browsers on PalmOS 4 were very limited even in their days, and are almost useless for today's web. Basic rendering, no CSS, no JavaScript and, most importantly, no HTTPS.

That said, you can use Eudora Web and Xiino (those are the ones that I and others have tested) to access and view plain HTML (mind the missing "S" in HTTPS 😏) pages, if you can find any of significance on today's web. Luckily, there's http://www.frogfind.com which rerenders pages into simple HTML and offloads SSL. With frogfind, you can indeed use your Palm to browse and read most of the stuff that you find.

**Comment by TheSystemGuy1337** [comment]
I guess I did it wrong. Even then, the modern web would for sure cause the Motorola Dragonball CPU (which I think is a 68K derivative) to melt through the device faster than a Prescott Pentium 4. Cloudpilot doesn’t emulate the later ARM based Palm Tungsten PDAs as far as I know.
